pkg/subctl/cmd/gather.go

```diff
@@ -266,3 +272,53 @@ func getAllModuleKeys() []string {
 	}
 	return keys
 }
+
+func createTar(sourcedir string, Status *cli.Status)  {
+	dir, err := os.Open(sourcedir)
+	if err != nil {
+		Status.QueueFailureMessage(fmt.Sprintf("error opening source directory %s", directory))
+	}
+	defer dir.Close()
+
+	files, err := dir.Readdir(0) // grab the files list
+	if err != nil {
+		if err != nil {
+			Status.QueueFailureMessage(fmt.Sprintf("error grabbing files from directory %s", dir))
+		}
+	}
+
+	tarfile, err := os.Create(sourcedir + ".tar") // Destination tarfile name
+	if err != nil {
+		Status.QueueFailureMessage(fmt.Sprintf("error creating destination tarfile %s", tarfile))
+	}
+	defer tarfile.Close()
+
+	tarfileWriter := tar.NewWriter(tarfile)
+	defer tarfileWriter.Close()
+
+	for _, fileInfo := range files {
+		// see https://www.socketloop.com/tutorials/go-file-path-independent-of-operating-system
+		file, err := os.Open(dir.Name() + string(filepath.Separator) + fileInfo.Name())
+		if err != nil {
+			Status.QueueFailureMessage(fmt.Sprintf("error opening file %s", file))
+		}
+		defer file.Close()
+
+		// prepare the tar header
+		header := &tar.Header{
+			Name: file.Name(),
+			Size: fileInfo.Size(),
+			Mode: int64(fileInfo.Mode())}
+
+		err = tarfileWriter.WriteHeader(header)
+		if err != nil {
+			Status.QueueFailureMessage("error writing tar header")
+		}
+
+		_, err = io.Copy(tarfileWriter, file)
+		if err != nil {
+			Status.QueueFailureMessage(fmt.Sprintf("error copying file %s to tar", file))
+		}
```

Comment:
It would be best to close each file after it's written to avoid keeping each file open at once. You can still use `defer` by putting this code in an anonymous function:

```suggestion
          err := func() error {
		defer file.Close()

		// prepare the tar header
		header := &tar.Header{
			Name: file.Name(),
			Size: fileInfo.Size(),
			Mode: int64(fileInfo.Mode())}

		err = tarfileWriter.WriteHeader(header)
		if err != nil {
		        return fmt.Sprintf("Error writing tar header: %v", err)
		}

		_, err = io.Copy(tarfileWriter, file)
		if err != nil {
		        return fmt.Sprintf("Error copying file %q to tar: %v", file.Name(), err)
		}
		
		return nil
	  }()
	  
	  if err != nil {
	      Status.QueueFailureMessage(err.Error())
	      return
	  }	
```